openssl/openssl.go

```diff
@@ -184,24 +199,47 @@ type fail string
 
 func (e fail) Error() string { return "boringcrypto: " + string(e) + " failed" }
 
+const wordBytes = bits.UintSize / 8
+
+// Reverse each limb of z.
+func (z BigInt) byteSwap() {
+	for i, d := range z {
+		var n uint = 0
+		for j := 0; j < wordBytes; j++ {
+			n |= uint(byte(d)) << (8 * (wordBytes - j - 1))
+			d >>= 8
+		}
+		z[i] = n
+	}
+}
+
 func wbase(b BigInt) *C.uint8_t {
 	if len(b) == 0 {
 		return nil
 	}
 	return (*C.uint8_t)(unsafe.Pointer(&b[0]))
 }
 
-const wordBytes = bits.UintSize / 8
-
 func bigToBN(x BigInt) *C.GO_BIGNUM {
+	if nativeEndian == binary.BigEndian {
+		z := x
```

Comment:
is this shallow copy necessary?

Reply:
It's not; removed.

Reply:
Actually, we might need a (deep) copy of it, as this modifies the original array passed by reference. I guess it is not exposed through tests, just because the input variables are used only once.